lib/plaid_handler.ex

```diff
@@ -1,34 +1,39 @@
-defmodule Plaid.Utils do
+defmodule PlaidHandler do
   @moduledoc """
-  Utility functions.
+  Maps external responses from Plaid's API to internal data structures.
   """
 
-  @type response :: %{required(String.t()) => any}
   @type endpoint :: atom
 
   @doc """
   Handles Plaid response and maps to the correct data structure.
   """
-  @spec handle_resp({:ok, HTTPoison.Response.t()} | {:error, HTTPoison.Error.t()}, endpoint) ::
-          {:ok, any} | {:error, Plaid.Error.t() | HTTPoison.Error.t()}
-  def handle_resp({:ok, %HTTPoison.Response{status_code: code} = resp}, endpoint)
+  @spec handle_resp({:ok, PlaidHTTP.Response.t()} | {:error, PlaidHTTP.Error.t()}, endpoint) ::
+          {:ok, term} | {:error, Plaid.Error.t() | PlaidHTTP.Error.t()} | no_return
+  def handle_resp({:ok, %PlaidHTTP.Response{body: body}}, _endpoint) when not is_map(body) do
+    raise PlaidHTTP.Error,
+      message: """
+        PlaidHTTP.call/5 must return a PlaidHTTP.Response struct with a map data type
+        for the key :body.
+
+        Value for :body was #{inspect(body)}
+      """
+  end
+
+  def handle_resp({:ok, %PlaidHTTP.Response{status_code: code, body: body}}, endpoint)
       when code in 200..201 do
```

Comment:
```suggestion
      when code in 200..299 do
```

Wouldn't successes be all the way to 299?

Reply:
Yes, iirc the `200..201` was based on empirical codes sent by Plaid. (It event sends `201` when tokens are created.) I believe my intent was to document the way the Plaid API behaved and let any exceptions error. Either way, good change and I implemented it.